crates/bevy_math/src/bounding/bounded3d/extrusion.rs

```diff
@@ -83,217 +74,186 @@ impl Bounded3d for Extrusion<Line2d> {
+            half_height: half_depth,
+            radius: self.radius,
         }
         .aabb_3d(Vec3::ZERO, rotation * Quat::from_rotation_x(FRAC_PI_2));
 
-        let up = rotation * Vec3::new(0., self.base_shape.half_length, 0.);
+        let up = rotation * Vec3::new(0., self.half_length, 0.);
         let half_size = Into::<Vec3>::into(aabb.max) + up.abs();
         Aabb3d::new(translation, half_size)
     }
+}
+
+impl<T: BoundedExtrusion> Bounded3d for Extrusion<T> {
+    fn aabb_3d(&self, translation: Vec3, rotation: Quat) -> Aabb3d {
+        self.base_shape
+            .extrusion_aabb_3d(self.half_depth, translation, rotation)
+    }
 
     fn bounding_sphere(&self, translation: Vec3, rotation: Quat) -> BoundingSphere {
-        extrusion_bounding_sphere(self, translation, rotation)
+        self.base_shape
+            .extrusion_bounding_sphere(self.half_depth, translation, rotation)
     }
 }
 
-/// Computes the axis aligned bounding box ([`Aabb3d`]) for an extrusion given its translation and rotation.
-pub fn extrusion_bounding_box<T: Primitive2d + Bounded2d>(
-    extrusion: &Extrusion<T>,
-    translation: Vec3,
-    rotation: Quat,
-) -> Aabb3d {
-    let cap_normal = rotation * Vec3::Z;
-    let conjugate_rot = rotation.conjugate();
-
-    // The `(halfsize, offset)` for each axis
-    let axis_values = Vec3::AXES.map(|ax| {
-        // This is the direction of the line of intersection of a plane with the `ax` normal and the plane containing the cap of the extrusion.
-        let intersect_line = ax.cross(cap_normal);
-        if intersect_line.length_squared() <= f32::EPSILON {
-            return (0., 0.);
-        };
-
-        // This is the normal vector of the intersection line rotated to be in the XY-plane
-        let line_normal = (conjugate_rot * intersect_line).yx();
-        let angle = line_normal.to_angle();
-
-        // Since the plane containing the caps of the extrusion is not guaranteed to be orthgonal to the `ax` plane, only a certain "scale" factor
-        // of the `Aabb2d` will actually go towards the dimensions of the `Aabb3d`
-        let scale = cap_normal.reject_from(ax).length();
-
-        // Calculate the `Aabb2d` of the base shape. The shape is rotated so that the line of intersection is parallel to the Y axis in the `Aabb2d` calculations.
-        // This guarantees that the X value of the `Aabb2d` is closest to the `ax` plane
-        let aabb2d = extrusion.base_shape.aabb_2d(Vec2::ZERO, angle);
-        (aabb2d.half_size().x * scale, aabb2d.center().x * scale)
-    });
-
-    let offset = Vec3A::from_array(axis_values.map(|(_, offset)| offset));
-    let cap_size = Vec3A::from_array(axis_values.map(|(max_val, _)| max_val)).abs();
-    let depth = rotation * Vec3A::new(0., 0., extrusion.half_depth);
-
-    Aabb3d::new(Vec3A::from(translation) - offset, cap_size + depth.abs())
-}
+/// A trait that automatically implements `Bounded3d` for `Extrusion<Self>`.
```

Comment:
This is good as a motivation, but I think it could do a better job of illustrating how it is used. e.g.:
```suggestion
/// A trait implemented on 2D shapes which determines the 3D bounding volumes of their extrusions.
/// 
/// Since default implementations can be inferred from 2D bounding volumes, this allows a `Bounded2d`
/// implementation on some shape `MyShape` to be extrapolated to a `Bounded3d` implementation on
/// `Extrusion<MyShape>` without supplying any additional data; e.g.:
/// `impl BoundedExtrusion for MyShape {}`
```